### PR TITLE
Added test case to validate security flavor

### PR DESCRIPTION
**Pull request by Krishan-Saraswat**
localhost:/home/avocado-misc-tests/security # avocado run --max-parallel-tasks=1 security_flavor.py
JOB ID     : cd20d3057efc282bbc85aed93345a45fe2d60f53
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-01-22T21.21-cd20d30/job.log
 (1/1) security_flavor.py:SecurityFlavor.test_security_flavor: STARTED
 (1/1) security_flavor.py:SecurityFlavor.test_security_flavor: PASS (0.01 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-01-22T21.21-cd20d30/results.html
JOB TIME   : 23.05 s

### CODE REVIEW

File: security/security_flavor.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        Security flavor values according to IBM support page
+        0 Speculative execution fully enabled
+        1 Speculative execution controls to mitigate user-to-kernel attacks
+        2 Speculative execution controls to mitigate user-to-kernel and
+        user-to-user side-channel attacks
+        '''
+        if not os.path.isfile("/proc/powerpc/lparcfg"):
+            self.fail("lparcfg file doesn't exist")
+        lparcfg = "/proc/powerpc/lparcfg"
+        if not os.path.isfile(lparcfg):
```
Comment (Naresh-ibm) [inline security/security_flavor.py:37]: @Krishan-Saraswat    `lparcfg = "/proc/powerpc/lparcfg"`
and ` lparcfg` both are same. is this some typo error?
please check

File: security/security_flavor.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            self.fail("lparcfg proc file doesn't exist")
+        cmd = "lparstat -x"
+
+        output = genio.read_file(lparcfg)
```
Comment (Naresh-ibm) [inline security/security_flavor.py:41]: is the `output` single line or multiple line?
if it is multiple line I feel we have run it with a loop. please re-check

File: security/security_flavor.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        lparcfg_value = output.splitlines()[-1].strip().split("=")[-1]
+        lparstat_output = process.system_output(cmd, ignore_status=True).decode()
+        lparstat_value = lparstat_output.split(":")[-1].strip()
```
Comment (Naresh-ibm) [inline security/security_flavor.py:44]: same here

**Comment by Naresh-ibm** [comment]
@Krishan-Saraswat any update?

File: security/security_flavor.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    def test_security_flavor(self):
+        '''
+        Security flavor values according to IBM support page
+        0 Speculative execution fully enabled
+        1 Speculative execution controls to mitigate user-to-kernel attacks
+        2 Speculative execution controls to mitigate user-to-kernel and
+        user-to-user side-channel attacks
+        '''
+        if not os.path.isfile("/proc/powerpc/lparcfg"):
+            self.fail("lparcfg file doesn't exist")
```
Comment (PraveenPenguin) [inline security/security_flavor.py:35]: should not be  cancel here?

File: security/security_flavor.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        lparcfg = "/proc/powerpc/lparcfg"
+        if not os.path.isfile(lparcfg):
+            self.fail("lparcfg proc file doesn't exist")
```
Comment (PraveenPenguin) [inline security/security_flavor.py:38]: here as well

File: security/security_flavor.py (hunk 1 above)
Comment (Krishan-Saraswat) [reply security/security_flavor.py]: Removed the extra duplicate if condition

File: security/security_flavor.py (hunk 2 above)
Comment (Krishan-Saraswat) [reply security/security_flavor.py]: output is multiple lines, after splitting the lines I iterated a loop to fetch the security flavor

File: security/security_flavor.py (hunk 3 above)
Comment (Krishan-Saraswat) [reply security/security_flavor.py:45]: "lparstat -x" command is specific to get the "Security mode value" so it's a single-line output only therefore it's fine.

File: security/security_flavor.py (hunk 4 above)
Comment (Krishan-Saraswat) [reply security/security_flavor.py]: Resolved

File: security/security_flavor.py (hunk 5 above)
Comment (Krishan-Saraswat) [reply security/security_flavor.py]: Removed this as it was duplicate

**Review by PraveenPenguin — APPROVED** [review]

**Review by Naresh-ibm — APPROVED** [review]
LGTM

**Pull request merged**